app/lib/actions/pulp/repository/presenters/puppet_presenter.rb

```diff
@@ -25,11 +25,14 @@ def humanized_details
           end
 
           def finished_count
-            task_result['finished_count']
+            return 0 unless task_progress_details.key?('modules')
+            task_progress_details['modules']['finished_count']
           end
 
           def total_count
-            task_progress_details['modules']['total_count'] || 0
+            return 0 unless task_progress_details.key?('modules')
+            return 0 if task_progress_details['modules']['total_count'].nil?
```

Comment:
Not sure I'm crazy about the dual return statements.  I wish rails would monkeypatch Hash with something for this.

What about fetch or try?

```ruby
return 0 if task_progress_details.fetch('modules', {})['total_count'].blank?
```

or

```ruby
return 0 if task_progress_details['modules'].try(:[], 'total_count').blank?
```